### PR TITLE
List all languages that we believe work on Spin

### PR DESCRIPTION
**Pull request by technosophos**
Add a list of languages that we believe will currently work on Spin

Signed-off-by: Matt Butcher <matt.butcher@fermyon.com>

### CODE REVIEW

File: docs/content/other-languages.md [hunk 1]
```diff
@@ -34,3 +34,71 @@ build and run Spin components in that programming language
...
+
+## Grain
+
+[Grain](https://grain-lang.org/), a new functional programming language, has WASI/Wagi support and can be used to write Spin apps.
+
+- The [Grain entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/grain) has details
+- A simple [Hello World example](https://github.com/deislabs/hello-wagi-grain) shows how to use Grain
+- For a production-quality example. the [Wagi Fileserver](https://github.com/deislabs/wagi-fileserver) is written in Grain
+
+## Haskell
```
Comment (itowlson) [inline docs/content/other-languages.md:70]: I would counsel against mentioning this unless you have had a lot more luck building it than I have.  Running one of their generated modules under the `wasmtime` CLI gives me _unknown import: `StablePtr::newStablePtr` has not been defined_.  Supposedly the Wasm can be run under Node but that just gives me a different error.

Or we could say "they _claim_ this works but we have never tried it."  Just something super non committal!

File: docs/content/other-languages.md [hunk 2]
```diff
@@ -34,3 +34,71 @@ build and run Spin components in that programming language
...
+## Ruby
+
+Upstream [Ruby](https://www.ruby-lang.org/en/) officially supports WebAssembly and WASI, and we here at Fermyon have successfully run Ruby apps in Spin.
+
+- The [Ruby entry in the Wasm Language Guide](https://www.fermyon.com/wasm-languages/ruby) has the latest information
+- [Ruby's 3.2.0 Preview 1 release notes](https://www.ruby-lang.org/en/news/2022/04/03/ruby-3-2-0-preview1-released/) detail WASI support
+
+## Zig
+
+Zig is a low-lelve systems language that has support for Wasm and WASI, and can be used to write Spin apps.
```
Comment (itowlson) [inline docs/content/other-languages.md:101]: typo 
```suggestion
Zig is a low-level systems language that has support for Wasm and WASI, and can be used to write Spin apps.
```

File: docs/content/other-languages.md (hunk 1 above)
Comment (technosophos) [reply docs/content/other-languages.md]: I tried it... and I can't figure out how to make it work. So I removed.

**Comment by technosophos** [comment]
I also removed the JavaScript section after trying both Javys and WasmEdge's QuickJS -- none of which work on `wasmtime` even.

**Review by lann — APPROVED** [review]

**Pull request merged**